entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/config/EntityUpdateJobFactory.java

```diff
@@ -0,0 +1,171 @@
+                .writer((ItemWriter<BatchEntityRecord>) applicationContext.getBean(compositeEntityInsertionWriter))
+                .listener((ItemProcessListener<? super BatchEntityRecord, ? super BatchEntityRecord>) itemListener)
+                .faultTolerant()
+                .skipPolicy(noopSkipPolicy)
+                .taskExecutor((TaskExecutor)applicationContext.getBean(WEB_REQUEST_JOB_EXECUTOR))
+                .throttleLimit(emConfig.getBatchUpdatesThrottleLimit())
+                .listener(stepExecutionListener(List.of(ScheduledUpdateType.FULL_UPDATE), true))
+                .build();
+
+    }
+
+
+    /**
+     * Creates Item reader based on synchronus and async updates
+     * @param synchronous
+     * @return
+     */
+    private ItemReader<BatchEntityRecord> getReader(boolean synchronous) {
+        return synchronous ? (EntityRecordDatabaseReader) applicationContext.getBean(SINGLE_ENTITY_RECORD_READER)
+                : (SynchronizedItemStreamReader)applicationContext.getBean(SCHEDULED_TASK_READER);
+    }
+
+
+    /**
+     * Creating it as a bean as the processor list is same for all the Internal Task of EM.
+     * For performnace will access them from application context than creating a list for every request
+     * @see <a href="http://docs.google.com/document/d/16k9PcCMFwl2LXjnnzotZRPc-QqM-Ar1D0VELHt4t_hA/edit?tab=t.0#heading=h.fj6e15rbq64q"></a> }
+     *
+     * Creates the processor list -
+     *    Processors: Dereference + consolidation + metrics + validation
+     * @return
+     */
+    @Bean(compositeProcessor)
+    private ItemProcessor<BatchEntityRecord, BatchEntityRecord> compositeUpdateProcessor() {
+        CompositeItemProcessor<BatchEntityRecord, BatchEntityRecord> compositeItemProcessor =
+                new CompositeItemProcessor<>();
+        compositeItemProcessor.setDelegates(
+                Arrays.asList(
+                        applicationContext.getBean(entityDereferenceProcessor, EntityDereferenceProcessor.class),
+                        applicationContext.getBean(entityConsolidationProcessor, EntityConsolidationProcessor.class),
+                        applicationContext.getBean(entityMetricsProcessor, EntityMetricsProcessor.class),
+                        applicationContext.getBean(entityVerificationLogger, EntityVerificationLogger.class)));
+        return compositeItemProcessor;
+    }
+
+    /**
+     * Creating it as a bean as the writer list is same for all the Internal Task of EM.
+     * For performance will access them from application context than creating a list for every request
+     * @see <a href="http://docs.google.com/document/d/16k9PcCMFwl2LXjnnzotZRPc-QqM-Ar1D0VELHt4t_hA/edit?tab=t.0#heading=h.fj6e15rbq64q"></a> }
+     * Creates the writer list -
+     *    Writer: Db update + Solr update
+     * @return
+     */
+    @Bean(compositeEntityInsertionWriter)
```

Comment:
The item writers are always the same, we should change the bean name toentityUpdateWriters, and instantiate them in Autoconfig class, the factory method should take the list with the names of the individual writters. Also the entityRecordDBInsertionWriter, and entitySolrInsertionWriter should be instantiated in the Autoconfig.